fedimint-server/src/consensus/engine.rs

```diff
@@ -472,6 +480,79 @@ impl ConsensusEngine {
             .expect("This is the only place where we write to this key");
     }
 
+    /// Creates a backup of the database in the checkpoint directory. The
+    /// frequency of the checkpoints is controlled by
+    /// `FM_DB_CHECKPOINT_FREQ_ENV`. These checkpoints can be used to
+    /// restore the database in case the federation falls out of consensus.
+    fn checkpoint_database(&self, session_index: u64) {
+        // By default, checkpoint the database every 10 sessions
+        let checkpoint_frequency: u64 = env::var(FM_DB_CHECKPOINT_FREQ_ENV)
+            .unwrap_or(10.to_string())
+            .parse()
+            .expect("FM_DB_CHECKPOINT_FREQ var is invalid");
+
+        if session_index % checkpoint_frequency == 0 {
+            let checkpoint_dir = self.data_dir.join("db_checkpoints");
+
+            if !checkpoint_dir.exists() {
+                if let Err(e) = fs::create_dir_all(&checkpoint_dir) {
```

Comment:
Would make it once somewhere, unconditionally to avoid having to check over and over.